File: lca/src/main/java/com/dreampany/lca/data/source/pref/LoadPref.java
package com.dreampany.lca.data.source.pref;

import android.content.Context;

import com.dreampany.frame.data.source.pref.BasePref;
import com.dreampany.frame.util.TimeUtil;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by Hawladar Roman on 10/2/2018.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 */
@Singleton
public class LoadPref extends BasePref {

    private static final String DEFAULT_FAVORITE_COMMITTED = "default_favorite_committed";
    private static final String COIN_INDEX_TIME = "coin_index_time";

    @Inject
    LoadPref(Context context) {
        super(context);
    }

    synchronized public void commitDefaultFavorite() {
        setPrivately(DEFAULT_FAVORITE_COMMITTED, true);
    }

    synchronized public boolean isDefaultFavoriteCommitted() {
        return getPrivately(DEFAULT_FAVORITE_COMMITTED, false);
    }

    synchronized public void setCoinIndexTime(int coinIndex) {
        setPrivately(COIN_INDEX_TIME + coinIndex, TimeUtil.currentTime());
    }

    synchronized public long getCoinIndexTime(int coinIndex) {
        return getPrivately(COIN_INDEX_TIME + coinIndex, 0L);
    }
}

File: tools/src/main/kotlin/com/dreampany/tools/data/source/pref/LoadPref.kt
package com.dreampany.tools.data.source.pref

import android.content.Context
import com.dreampany.framework.data.source.pref.FramePref
import com.dreampany.tools.misc.Constants
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by Roman-372 on 9/2/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
@Singleton
class LoadPref
@Inject constructor(
    context: Context
) : FramePref(context) {

    override fun getPrivatePrefName(context: Context): String? {
        return Constants.Pref.LOAD
    }
}